pkg/executor/composite_cache.go

```diff
@@ -85,16 +91,14 @@ func (s *CompositeCache) AddPath(p, context string) error {
 }
 
 // HashDir returns a hash of the directory.
-func HashDir(p, context string) (string, error) {
+func hashDir(p, context string) (bool, string, error) {
 	sha := sha256.New()
+	empty := true
 	if err := filepath.Walk(p, func(path string, fi os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
 		exclude := util.ExcludeFile(path, context)
-		if fi.IsDir() && exclude {
```

Comment:
This is still needed right so we can skip walking whitelisted directories?

Reply:
Actually this was onr if the bugs:
If the directory was ignored, but one the contained files was included via later rules in the ignore file, this would skip over this file completely.

The change now would still not include the directory in the cache key, but would include the contained file.